cdap-data-fabric/src/main/java/co/cask/cdap/data2/sql/SqlStructuredRow.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.sql;
+
+import co.cask.cdap.spi.data.InvalidFieldException;
+import co.cask.cdap.spi.data.StructuredRow;
+import co.cask.cdap.spi.data.table.StructuredTableSchema;
+import co.cask.cdap.spi.data.table.field.FieldType;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * The SqlStructuredRow represents one row in the sql table.
+ */
+public class SqlStructuredRow implements StructuredRow {
+
+  private final Map<String, Object> columns;
+  private final StructuredTableSchema tableSchema;
+
+  public SqlStructuredRow(StructuredTableSchema tableSchema) {
+    this.tableSchema = tableSchema;
+    this.columns = new HashMap<>();
+  }
+
+  public <T> void add(String columnName, T data) {
+    // only put if the field is not null
+    if (data != null) {
+      columns.put(columnName, data);
+    }
+  }
+
+  @Nullable
+  @Override
+  public Integer getInteger(String fieldName) throws InvalidFieldException {
+    validateField(fieldName, Collections.singleton(FieldType.Type.INTEGER));
```

Comment:
This validation is needed because user can accidentally put the wrong field name and call the wrong method to get the value, so I think the validation should be here.